src/hooks/useQuery.ts

```diff
@@ -6,22 +6,31 @@ import { useMemo } from 'react'
  *
  * @param value Input from URL
  */
-function sanitizeInput(value?: string | null): string {
-  return value && Number(value) ? value : '0'
+function sanitizeInput(value?: string | null, defaultValue = '0'): string {
+  return value && Number(value) ? value : defaultValue
 }
 
-export function useQuery(): { sellAmount: string; buyAmount: string } {
-  const query = new URLSearchParams(useLocation().search)
+/**
+ * Prevents invalid NEGATIVE numbers from being inserted by hand in the URL
+ *
+ * @param value Input from URL
+ */
+function sanitizeNegativeInput(value?: string | null, defaultValue = '0'): string {
+  return (Number(value) === 0 || Number(value) >= 5) && value ? value : defaultValue
```

Comment:
I don't understand this expression:
- Why 5?
- It looks `Number(value)` can be factorized
- Why `&& value` is checked after you actually use the `value`. Maybe you want to invert the order of the AND claused